urwid/widget/listbox.py

```diff
@@ -365,11 +373,14 @@ def body(self, body: Iterable[Widget] | ListWalker) -> None:
             self._body = SimpleListWalker(body)
         try:
             signals.connect_signal(self._body, "modified", self._invalidate)
+            signals.connect_signal(self._body, "modified", self._clear_cache)
```

Comment:
downside of this call is that child widgets can change their size and no notification will arrive (`_invalidate` will be called at such situation)

Reply:
that's true, i haven't considered that aspect